anaUltraScurve.py

```diff
@@ -154,7 +154,7 @@ def plotAllSCurvesOnCanvas(vfatHistos, vfatHistosPanPin2=None, obsName="scurves"
                       help="Physical filename of a custom, non-default, channel mapping (optional)", metavar="extChanMapping")
     parser.add_option("-f", "--fit", action="store_true", dest="performFit",
                       help="Fit scurves and save fit information to output TFile", metavar="performFit")
-    parser.add_option("--isVFAT3", action="store_true", dest="isVFAT3",
+    parser.add_option("--isVFAT3", action="store_true", dest="isVFAT3", default=False,
```

Comment:
Why set a default here? `store_true` defaults to `False` automatically

Reply:
It does not, it defaults to None.  This was found when passing it to the instance of ScanDataFitter and caused an exception to be generated due to a NoneType

Reply:
very strange... (strange that `None` and `False` if used as boolean variables don't behave the same for you)
which line is the failure traced to?

Reply:
It shows up right here:

https://github.com/cms-gem-daq-project/gem-plotting-tools/blob/a13112a626b7b532835a5f0b0dcc497b7f47e3ab/anaUltraScurve.py#L296-L300

If the user supplies `--isVFAT3` then `options.isVFAT3` here gives `True` to the constructor of `ScanDataFitter` however if the argument is not supplied and the default is not set `None` is given to the constructor which then throws an error if suplying the `--debug` option because:

https://github.com/cms-gem-daq-project/gem-plotting-tools/blob/a13112a626b7b532835a5f0b0dcc497b7f47e3ab/fitting/fitScanData.py#L226-L241

Reply:
Setting the `default` to `False` resolved this issue.

Reply:
ah, it's because you're using an `integer` string format for a `boolean` type (works for `True/False` but not, by design, `None`), formatting it as a `boolean` (`%r`) or casting the value to a `bool` in your current implementation would work in all cases